cni-plugin/test/install-cni_test.go

```diff
@@ -0,0 +1,360 @@
+		t.Fatalf("Failed to read file %v, err: %v", tempResult, err)
+	}
+
+	expectedFile, err := ioutil.ReadFile(expected)
+	if err != nil {
+		t.Fatalf("Failed to read file %v, err: %v", expected, err)
+	}
+
+	if bytes.Equal(resultFile, expectedFile) {
+		t.Logf("PASS: result matches expected: %v v. %v", tempResult, expected)
+	} else {
+		tempFail := mktemp(testWorkRootDir, result+".fail.XXXX", t)
+		t.Errorf("FAIL: result doesn't match expected: %v v. %v", tempResult, expected)
+		cp(tempResult, tempFail+"/"+result, t)
+		t.Fatalf("Check %v for diff contents", tempFail)
+	}
+}
+
+// checkBinDir verifies the presence/absence of test files.
+func checkBinDir(t *testing.T, tempCNIBinDir string, op string, files ...string) {
+	for _, f := range files {
+		if _, err := os.Stat(tempCNIBinDir + "/" + f); !os.IsNotExist(err) {
+			if op == "add" {
+				t.Logf("PASS: File %v was added to %v", f, tempCNIBinDir)
+			} else if op == "del" {
+				t.Fatalf("FAIL: File %v was not removed from %v", f, tempCNIBinDir)
+			}
+		} else {
+			if op == "add" {
+				t.Fatalf("FAIL: File %v was not added to %v", f, tempCNIBinDir)
+			} else if op == "del" {
+				t.Logf("PASS: File %v was removed from %v", f, tempCNIBinDir)
+			}
+		}
+	}
+}
+
+// doTest sets up necessary environment variables, runs the Docker installation
+// container and verifies output file correctness.
+func doTest(testNum int, wd string, initialNetConfFile string, finalNetConfFile string, expectNetConfFile string, expectedPostCleanNetConfFile string, tempCNINetDir string, tempCNIBinDir string, tempK8sSvcAcctDir string, testWorkRootDir string, t *testing.T) {
+	t.Logf("Test %v: prior cni-conf='%v', expected result='%v'", testNum, initialNetConfFile, finalNetConfFile)
+
+	if initialNetConfFile != "NONE" {
+		setEnv(cniConfName, initialNetConfFile, t)
+	}
+	defaultData, err := ioutil.ReadFile(wd + "../deployment/linkerd-cni.conf.default")
+	if err != nil {
+		t.Fatalf("Failed to read file %v, err: %v", wd+"../deployment/linkerd-cni.conf.default", err)
+	}
+	setEnv(cniNetworkConfigName, string(defaultData), t)
+
+	containerID := startDocker(testNum, wd, testWorkRootDir, tempCNINetDir, tempCNIBinDir, tempK8sSvcAcctDir, t)
+	time.Sleep(5 * time.Second)
+
+	compareConfResult(testWorkRootDir, tempCNINetDir, finalNetConfFile, expectNetConfFile, t)
+	checkBinDir(t, tempCNIBinDir, "add", "linkerd-cni")
+
+	docker("stop", containerID, t)
+	time.Sleep(5 * time.Second)
+
+	t.Logf("Test %v: Check the cleanup worked", testNum)
+	checkBinDir(t, tempCNIBinDir, "del", "linkerd-cni")
+	if len(expectedPostCleanNetConfFile) > 0 {
+		compareConfResult(testWorkRootDir, tempCNINetDir, finalNetConfFile, expectedPostCleanNetConfFile, t)
+	} else {
+		files := ls(tempCNINetDir, t)
+		if len(files) > 0 {
+			t.Fatalf("FAIL: CNI_CONF_DIR is not empty: %v", files)
+		} else {
+			t.Log("PASS: CNI_CONF_DIR is empty")
+		}
+	}
+
+	docker("logs", containerID, t)
+	docker("rm", containerID, t)
+}
+
+func TestMain(m *testing.M) {
```

Comment:
🎉 This test file passes for me now, when I add the `-integration-tests` flag! Thanks for setting that up.